cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -210,15 +275,15 @@ where
         Ok(match ty {
             ty if ty.is_int() => {
                 let imm64 = match ty {
-                    I8 => self.u.arbitrary::<i8>()? as i64,
-                    I16 => self.u.arbitrary::<i16>()? as i64,
-                    I32 => self.u.arbitrary::<i32>()? as i64,
-                    I64 => self.u.arbitrary::<i64>()?,
+                    ty if ty == I8 => self.u.arbitrary::<i8>()? as i64,
+                    ty if ty == I16 => self.u.arbitrary::<i16>()? as i64,
+                    ty if ty == I32 => self.u.arbitrary::<i32>()? as i64,
+                    ty if ty == I64 => self.u.arbitrary::<i64>()?,
                     _ => unreachable!(),
                 };
```

Comment:
This change is unnecessary, right? I'd like to keep the simpler syntax unless there's some reason it needs to change.

Reply:
My editor was complaining that we were always matching on I8 as a variable name and that the others were unused.

But looking at it again it probably would also generate a compile warning, so i think it might be a bug on CLion.

Ill revert this change.